src/Akka.Persistence.Sql/Query/Dao/BaseByteReadArrayJournalDao.cs

```diff
@@ -50,20 +52,19 @@ public abstract class BaseByteReadArrayJournalDao : BaseJournalDaoWithReadMessag
             var maxTake = MaxTake(max);
 
             return AsyncSource<string>.FromEnumerable(
-                new { _connectionFactory = ConnectionFactory, maxTake },
-                async input =>
+                new { _dbStateHolder, maxTake },
+                static async input =>
                 {
-                    return await input._connectionFactory.ExecuteWithTransactionAsync(
-                        ReadIsolationLevel,
-                        ShutdownToken,
-                        async (connection, token) =>
+                    return await input._dbStateHolder.ExecuteWithTransactionAsync(
+                        input.maxTake,
+                        async (connection, token,take) =>
```

Comment:
Can we just return a `Task` here instead of `async` / `await` ?

Reply:
One discussion on this is here: https://stackoverflow.com/questions/24110213/difference-between-returning-and-awaiting-a-task-in-an-async-method

Would the performance boost for each query execution be worth it?

Reply:
I tend to agree with @to11mtm on this one, returning a `Task` from inside a `using` block is quite dangerous because we don't know just how robust the `IDispose` is being implemented by the 3rd party API developers.